### PR TITLE
Add add_bid, withdraw_bid, upgradable contract.

### PR DESCRIPTION
**Pull request by zie1ony**
Added:
- `add_bid` endpoint
- `withdraw_bid` endpoint
- upgradable contract
- `auction_manager_package_hash` named key
- `auction_manager_access_token` named key

### CODE REVIEW

File: auction-manager/src/main.rs [hunk 1]
```diff
@@ -5,68 +5,117 @@ use contract::contract_api::{runtime, storage, system};
...
-    let (contract_hash, _) = storage::new_locked_contract(entry_points, None, None, None);
-    runtime::put_key("auction_manager_contract", contract_hash.into());
-    runtime::put_key(
-        "auction_manager_contract_hash",
-        storage::new_uref(contract_hash).into(),
+    let _ = storage::new_contract(
+        entry_points,
+        None,
+        Some(String::from("auction_manager_access_token")),
+        Some(String::from("auction_manager_package_hash")),
```
Comment (EdHastingsCasperLabs) [inline auction-manager/src/main.rs:114]: I think you need to swap lines 113 and 114.
```
pub fn new_contract(
    entry_points: EntryPoints,
    named_keys: Option<NamedKeys>,
    hash_name: Option<String>,
    uref_name: Option<String>,
) 
```

File: auction-manager/Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "auction-manager"
-version = "0.1.0"
-authors = ["astro019 <dmitry.yakov7@gmail.com>"]
+version = "0.1.1"
+authors = ["astro019 <dmitry.yakov7@gmail.com>", "Maciej Zieliński <maciej@casperlabs.io"]
```
Comment (EdHastingsCasperLabs) [inline auction-manager/Cargo.toml:4]: change to dmitry's casperlabs email?

**Review by EdHastingsCasperLabs — APPROVED** [review]
See comments, otherwise lgtm.

File: auction-manager/Cargo.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "auction-manager"
-version = "0.1.0"
-authors = ["astro019 <dmitry.yakov7@gmail.com>"]
+version = "0.1.1"
```
[the remark this answers is not in the document]
Comment (zie1ony) [reply auction-manager/Cargo.toml:3]: There's no breaking changes in endpoints, so I made smallest possible version increment.

File: auction-manager/src/main.rs (hunk 1 above)
Comment (zie1ony) [reply auction-manager/src/main.rs]: My bad. Thanks.

**Pull request merged**